src/main/java/com/conveyal/gtfs/model/Stop.java

```diff
@@ -125,4 +187,141 @@ public Iterator<Stop> iterator() {
+            stopsReader.get(ZONE_ID_FIELD) + "," +
+            stopsReader.get(STOP_URL_FIELD) + "," +
+            stopsReader.get(LOCATION_TYPE_FIELD) + "," +
+            stopsReader.get(PARENT_STATION_FIELD) + "," +
+            stopsReader.get(STOP_TIMEZONE_FIELD) + "," +
+            stopsReader.get(WHEELCHAIR_BOARDING_FIELD) + "," +
+            stopsReader.get(PLATFORM_CODE_FIELD) + "," +
+            stopAreaIds;
+    }
+
+    /**
+     * Convert the multiple stops (with stop areas) back into CSV, with header and return a {@link CsvReader}
+     * representation.
+     */
+    private static CsvReader produceCsvPayload(List<String> rows) {
+        StringBuilder csvContent = new StringBuilder();
+        csvContent.append(CSV_HEADER_FOR_MERGE);
+        rows.forEach(row -> csvContent.append(row).append(System.lineSeparator()));
+        return new CsvReader(new StringReader(csvContent.toString()));
+    }
+
+    /**
+     * Extract the stop areas from file and group by stop id. This is to allow for easier CRUD by the DT UI.
+     */
+    public static Map<String, Set<String>> groupStopAreaIds(CsvReader csvReader, List<String> errors) {
+        Map<String, Set<String>> stopAreasGroupedByStopId = new HashMap<>();
+
+        try {
+            while (csvReader.readRecord()) {
+                if (!hasExpectedNumberOfColumns(csvReader, errors, 2)) {
+                    continue;
+                }
+                String stopAreaId = csvReader.get(AREA_ID_FIELD);
+                String stopId = csvReader.get(STOP_ID_FIELD);
+                stopAreasGroupedByStopId.computeIfAbsent(stopId, k -> new HashSet<>()).add(stopAreaId);
+            }
+            return stopAreasGroupedByStopId;
+        } catch (IOException e) {
+            return Collections.emptyMap();
+        }
+    }
+
+    /**
+     * Expand the stop area ids and write to zip file.
+     */
+    public static void writeStopAreasToFile(ZipOutputStream zipOutputStream, List<Stop> stops) throws IOException {
+        // Create entry for table.
+        zipOutputStream.putNextEntry(new ZipEntry(STOP_AREAS_FILE_NAME));
+        // Create and use PrintWriter, but don't close. This is done when the zip entry is closed.
+        PrintWriter p = new PrintWriter(zipOutputStream);
+        p.print(packStopAreas(stops));
+        p.flush();
+        zipOutputStream.closeEntry();
+    }
+
+    /**
+     * Expand all stop area ids into a single row for each area id. This is to conform with the GTFS Fares v2 standard.
+     */
+    public static String packStopAreas(List<Stop> stops) {
+        StringBuilder csvContent = new StringBuilder(AREA_ID_FIELD)
+            .append(",")
+            .append(STOP_ID_FIELD)
+            .append(System.lineSeparator());
+
+        stops.stream()
+            .filter(stop -> stop.stop_area_ids != null)
+            .forEach(stop -> {
+                String[] areaIds = stop.stop_area_ids.split(";");
```

Comment:
What happens if the stop id has a semicolon in it?

Reply:
@miles-grant-ibigroup that would cause problems, similar to there being a comma. I can change this to something else, for instance a pipe (`|`).

Reply:
Could we do something stranger, like a custom unicode character or a sequence of pipes?